### PR TITLE
Add `Dockerfile` for steampipe and automated image deploy. Closes #662. Closes #677

### LINKED ISSUE #662: `steampipe` docker image

### LINKED ISSUE #677: `release` automation should also update `image` at Docker Hub

### PR DESCRIPTION
**Pull request by binaek89**

### CODE REVIEW

**Review by johnsmyth — APPROVED** [review]
@binaek89 I made some additional changes and pushed them to the branch.  lmk if you have any issue with my changes, else good to merge as far as im concerned

**Comment by binaek89** [comment]
This PR (temporarily) disables `acceptance` tests on PR. DO NOT merge this till that is removed.

File: .github/workflows/publish_docker.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Steampipe Publish Release
+on:
+  pull_request:
```
Comment (binaek89) [inline .github/workflows/publish_docker.yml:3]: This should change to `workflow_dispatch`

File: .github/workflows/publish_docker.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          build-args: |
+            TARGETOS=linux
+            TARGETARCH=amd64
+            TARGETVERSION=${{ env.STEAMPIPE_VERSION }}
+          tags: |
+            binaek/steampipe:${{ steps.generate_docker_tag.outputs.docker_tag }}
```
Comment (binaek89) [inline .github/workflows/publish_docker.yml:35]: This should be `turbot/steampipe:xxx`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -169,6 +168,37 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  # publish_docker_image:
```
Comment (kaidaguerre) [inline .github/workflows/release.yml:171]: what is this?

**Pull request merged**